processing/src/main/java/org/apache/druid/segment/data/BlockLayoutColumnarLongsSupplier.java

```diff
@@ -204,7 +204,10 @@ protected void loadBuffer(int bufferNum)
     public void close()
     {
       if (holder != null) {
+        currBufferNum = -1;
         holder.close();
+        holder = null;
+        buffer = null;
```

Comment:
nit: I guess `currBufferNum` is the only thing that truly matters here, but just to be consistent maybe we should also null out `longBuffer` since we are doing that for everything else.

Additionally, while I am not aware of anywhere which is double closing any float or double columns, they do suffer from the same flaw of not resetting `currBufferNum` when closed, and so could also read from a buffer that has been returned to the pool if that happened,
* https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/data/BlockLayoutColumnarFloatsSupplier.java#L183
* https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/data/BlockLayoutColumnarDoublesSupplier.java#L183

Reply:
+1 for fixing the other classes too.

Reply:
Thanks for pointing that out, fixed other supplier classes too.